cpu/cc2538/dev/uart.c

```diff
@@ -136,7 +136,9 @@
  * Baud rate defines used in uart_init() to set the values of UART_IBRD and
  * UART_FBRD in order to achieve the configured baud rates.
  */
-#define UART_CLOCK_RATE       16000000 /* 16 MHz */
+
+#define UART_CLOCK_RATE			(SYS_CTRL_32MHZ/(1 << SYS_CTRL_SYS_DIV))
+
```

Comment:
Which would also mean (Page 414:) 
"Note: If the PIOSC is used for the UART baud clock, the system clock frequency must be at  least 9 MHz in run mode." 
What's the reason for choosing the I/O clock?